framework/test/unit/modules/chain/unit/modules/blocks/process.js

```diff
@@ -1661,200 +1666,182 @@ describe('blocks/process', async () => {
 	});
 
 	describe('generateBlock', async () => {
-		describe('modules.transactions.getUnconfirmedTransactionList', async () => {
-			describe('when query returns empty array', async () => {
-				beforeEach(() => {
+		describe('modules.transactions.getUnconfirmedTransactionList', () => {
+			describe('when query returns empty array', () => {
+				beforeEach(async () => {
 					modules.transactions.getUnconfirmedTransactionList.returns([]);
-					return modules.blocks.verify.processBlock.callsArgWith(
+					modules.processTransactions.verifyTransactions.returns(
+						Promise.resolve([])
+					);
+					modules.blocks.verify.processBlock.callsArgWith(
 						3,
 						null,
 						modules.blocks.verify.processBlock.args
 					);
 				});
 
-				it('should generate block without transactions', done => {
+				it('should generate block without transactions', async () => {
```

Comment:
We are not returning a promise from this function, and the assertions in this test are defined in the callback. So, I think this test case finishes before the assertions are reached. Are you sure this is a valid test?

Reply:
Yes it is reaching the assertions properly (changing expected values fails test). But I can use `done` for better clarity.